src/cloudformation_cli_python_lib/interface.py

```diff
@@ -59,7 +59,31 @@ class HandlerErrorCode(str, _AutoName):
 
 class BaseResourceModel:
     def _serialize(self) -> Mapping[str, Any]:
-        return self.__dict__
+        ser = self.__dict__
+        del_keys = []
+        for k, v in ser.items():
```

Comment:
Something like this 
```
return { k: self.serialize_item(v)  for k, v in x.__dict__.items() if v is not None} 
```
might read cleaner with serialize_item being the contents of the loop w/o the delete logic.